src/remixAppManager.js

```diff
@@ -1,75 +1,77 @@
 /* global localStorage, fetch */
-import { PluginEngine, IframePlugin } from '@remixproject/engine'
+import { PluginManager, IframePlugin } from '@remixproject/engine'
 import { EventEmitter } from 'events'
-import { PermissionHandler } from './app/ui/persmission-handler'
 import QueryParams from './lib/query-params'
 
 const requiredModules = [ // services + layout views + system views
-  'compilerArtefacts', 'compilerMetadata', 'contextualListener', 'editor', 'offsetToLineColumnConverter', 'network', 'theme', 'fileManager', 'contentImport',
+  'manager', 'compilerArtefacts', 'compilerMetadata', 'contextualListener', 'editor', 'offsetToLineColumnConverter', 'network', 'theme', 'fileManager', 'contentImport',
   'mainPanel', 'hiddenPanel', 'sidePanel', 'menuicons', 'fileExplorers',
   'terminal', 'settings', 'pluginManager']
 
-const settings = {
-  permissionHandler: new PermissionHandler(),
-  autoActivate: false,
-  natives: ['vyper', 'workshops', 'ethdoc', 'etherscan'] // Force iframe plugin to be seen as native
+export function isNative (name) {
+  const nativePlugins = ['vyper', 'workshops', 'ethdoc', 'etherscan']
+  return nativePlugins.includes(name) || requiredModules.includes(name)
 }
 
-export class RemixAppManager extends PluginEngine {
+export class RemixAppManager extends PluginManager {
 
   constructor (plugins) {
-    super(plugins, settings)
+    super()
     this.event = new EventEmitter()
-    this.registered = {}
     this.pluginsDirectory = 'https://raw.githubusercontent.com/ethereum/remix-plugins-directory/master/build/metadata.json'
     this.pluginLoader = new PluginLoader()
   }
 
-  onActivated (plugin) {
+  async canActivate (from, to) {
```

Comment:
Maybe add a comment to explain what it does and a todo to have more precise behavior here.

Reply:
I'll change this in next PR